frappe/public/js/frappe/router.js

```diff
@@ -216,14 +216,13 @@ frappe.router = {
 			} else {
 				route = ["List", doctype_route.doctype, "List"];
 			}
-		} else if (frappe.model.is_single(doctype_route.doctype)) {
-			route = ["Form", doctype_route.doctype, doctype_route.doctype];
-		} else {
-			route = ["List", doctype_route.doctype, "List"];
-		}
-		// reset the layout to avoid using incorrect views
-		this.doctype_layout = doctype_route.doctype_layout;
-		return route;
+			if (doctype_route.doctype_layout) {
+				// set the layout
+				this.doctype_layout = doctype_route.doctype_layout;
+			}
```

Comment:
@shariquerik, this was changed in #18020

Reply:
Yes, I was about to update